tests/integration-tests/tests/storage/kms_key_factory.py

```diff
@@ -0,0 +1,207 @@
+        return policy_arn
+
+    def _create_kms_key(self, scheduler):
+        # create KMS key
+        self.kms_client = boto3.client("kms")
+        if scheduler == "awsbatch":
+            key_alias = "alias/Integration_test_KMS_key_batch_{0}".format(self.region)
+        else:
+            key_alias = "alias/Integration3_test_KMS_key_{0}".format(self.region)
+        # If the key already existed, use the existing key
+        for alias in self.kms_client.list_aliases().get("Aliases"):
+            if alias.get("AliasName") == key_alias:
+                kms_key_id = alias.get("TargetKeyId")
+                logging.info("Use existing KMS key {0}".format(kms_key_id))
+                return kms_key_id
+
+        # if the key doesn't existed in the account, create a new key
+        logging.info("Creating KMS key...")
+        response = self.kms_client.create_key(
+            Description="create kms key",
+            KeyUsage="ENCRYPT_DECRYPT",
+            Origin="AWS_KMS",
+            BypassPolicyLockoutSafetyCheck=False,
+        )
+        kms_key_id = response["KeyMetadata"]["KeyId"]
+
+        # create KMS key policy
+        key_policy = {
+            "Version": "2012-10-17",
+            "Statement": [
+                {
+                    "Sid": "Enable IAM User Permissions",
+                    "Effect": "Allow",
+                    "Principal": {"AWS": "arn:aws:iam::{account_id}:root".format(account_id=self.account_id)},
+                    "Action": "kms:*",
+                    "Resource": "*",
+                },
+                {
+                    "Sid": "Allow use of the key",
+                    "Effect": "Allow",
+                    "Principal": {
+                        "AWS": "arn:aws:iam::{account_id}:role/{iam_role_name}".format(
+                            account_id=self.account_id, iam_role_name=self.iam_role
+                        )
+                    },
+                    "Action": [
+                        "kms:Encrypt",
+                        "kms:Decrypt",
+                        "kms:ReEncrypt*",
+                        "kms:GenerateDataKey*",
+                        "kms:DescribeKey",
+                    ],
+                    "Resource": "*",
+                },
+                {
+                    "Sid": "Allow attachment of persistent resources",
+                    "Effect": "Allow",
+                    "Principal": {
+                        "AWS": "arn:aws:iam::{account_id}:role/{iam_role_name}".format(
+                            account_id=self.account_id, iam_role_name=self.iam_role
+                        )
+                    },
+                    "Action": ["kms:CreateGrant", "kms:ListGrants", "kms:RevokeGrant"],
+                    "Resource": "*",
+                    "Condition": {"Bool": {"kms:GrantIsForAWSResource": "true"}},
+                },
+            ],
+        }
+
+        # attach key policy to the key
+        time.sleep(20)
```

Comment:
why sleep here?

Reply:
I remember that kms_client.create_key take some time, if I don't sleep here, there may be error when execute put_key_policy, but I don't know what's the appropriate sleep time here.

Reply:
Could we use the boto3 kms client's [describe_key](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/kms.html#KMS.Client.describe_key) method to wait for the `KeyState` to reach the right value?

Reply:
After our discussion, it take time for the iam role to become valid, so I leave the sleep there and add the comments